product/engineering/JAVASCRIPT.md

```diff
@@ -53,4 +53,42 @@ avoid:
 
   * `.after()`, `.append()`, `.attr()`, `.before()`, `.clone()`, `.css()`, `.remove()`, and everything else.
 
+### When traversing the DOM many times, load one into memory first!
+```
+<div id="one">
+  <div id="two">
+    <div id="three">
+      <div id="four">
+      <div>
+    <div>
+  <div>
+<div>
+```
+
 
+```
+// BAD
+const one = $("#one");
+const two = $("#two");
+const three = $("#three");
+const four = $("#four");
+
+// GOOD
```

Comment:
can you elaborate on why this is preferred?

Reply:
Its less expensive to query the DOM once over four times.

Reply:
https://jsperf.com/selector-vs-find-again/11 Heres the performance test. See the speed for cache.find

Reply:
you should add it to the file :blush:

Reply:
Haha ok!